client/my-sites/sites/sites.jsx

```diff
@@ -139,100 +31,56 @@ module.exports = React.createClass( {
 			path = '/sites';
 		}
 
-		if ( this.state.search ) {
-			sites = this.props.sites.search( this.state.search );
-		} else {
-			sites = this.props.sites.getVisible();
-		}
-
-		/**
-		 * Filters sites based on public or private nature
-		 * for paths `/public` and `/private` only
-		 */
-		if ( path !== '/sites' && this.checkPathIsAFilter( this.getFilters() ) ) {
-			is_private = path === '/sites/private';
-			sites = sites.filter( function( site ) {
-				return site.is_private === is_private;
-			} );
+		// Filters sites based on public or private nature
+		// for paths `/public` and `/private` only
+		if ( path === '/sites/private' ) {
+			return site.is_private;
 		}
 
-		// filter out jetpack sites when on particular routes
+		// Filter out jetpack sites when on particular routes
 		if ( /^\/menus/.test( path ) || /^\/customize/.test( path ) ) {
-			sites = sites.filter( function( site ) {
-				return ! site.jetpack;
-			} );
+			return ! site.jetpack;
 		}
 
-		// filter out sites with no upgrades on particular routes
+		// Filter out sites with no upgrades on particular routes
 		if ( /^\/domains/.test( path ) || /^\/plans/.test( this.props.sourcePath ) ) {
-			sites = sites.filter( function( site ) {
-				return site.isUpgradeable();
-			} );
+			return site.isUpgradeable();
 		}
 
-		return sites;
+		return site;
 	},
 
-	render: function() {
-		var sites = this.sitesList(),
-			sitesMarkup,
-			siteSelectionHeaderText = this.translate( 'Please Select a Site:' ),
-			sitesTruncated;
-
-		if ( this.props.getSiteSelectionHeaderText ) {
-			siteSelectionHeaderText = this.props.getSiteSelectionHeaderText();
-		}
-
-		sitesTruncated = sites.slice( 0, this.state.page * this.props.perPage );
-
-		sitesMarkup = sitesTruncated.map( function( site ) {
-			return (
-				<SiteCard site={ site } key={ site.ID } sourcePath={ this.props.sourcePath }/>
-			);
-		}, this );
+	onSiteSelect: function( slug ) {
+		page.replace( addSiteFragment( this.props.path, slug ) );
+	},
 
-		if ( sites.length === sitesTruncated.length ) {
-			sitesMarkup.push( this.newSiteBox() );
-		}
+	getHeaderText() {
+		const path = this.props.path.replace( /\//g, ' ' );
+		return (
+			<span>
+				{ this.translate( 'Please select a site to open' ) }<strong>{ path }</strong>
+			</span>
+		)
+	},
 
+	render: function() {
 		return (
 			<Main className="sites">
-				<h2 className="sites__select-heading">{ siteSelectionHeaderText }</h2>
-				{ this.getSiteCount() > 1 ?
-					<SearchCard onSearch={ this.doSearch } analyticsGroup="Sites" />
-					: null
-				}
-				<div className="sites__grid">
-					{ sites.length !== 0 ? sitesMarkup : this.noSitesRender() }
-				</div>
+				<h2 className="sites__select-heading">
+					{ this.props.getSiteSelectionHeaderText
```

Comment:
Maybe I'm especially averse to ternaries, but seems like the `this.props.getSiteSelectionHeaderText` could be an early return condition of `getHeaderText`.